### DISCUSSION TITLE
latency and memory communication between python and javascript

### DISCUSSION THREAD

**Discussion by 5hayanB** [r0x0r/pywebview#1541] [Q&A] [answered]
My GUI requires showing a file tree. I wanted to know which of the following scenarios is more optimized way of doing the said task for a folder with many files and sub-directories.
1. Create nested dictionary of the folder structure in Python and pass that dictionary to JavaScript. Are sequence objects and maps copied between Python and JavaScript or only pointers to those sequences and maps are passed?
2. Repeatedly call a generator from Python through a loop in JavaScript and render it on the fly. Is invoking functions between Python and JavaScript an expensive operation? How much of a latency is there?

**Comment by r0x0r** [comment, marked as answer]
1. Javascript and Python are completely separated. Data is copied between two. If you pass data from Javascript to Python via JSAPI, data objects are serialized into JSON, passed to Python and deserialized to Python objects. The format of data passed from Python to JS via `evaluate_js` is up to you, but JSON is probably the best choice
2. I have not done any profiling, but there is a current issue touching this https://github.com/r0x0r/pywebview/issues/1545
